app/src/main/java/lt/vilnius/tvarkau/mvp/interactors/ReportTypesInteractorImpl.kt

```diff
@@ -0,0 +1,34 @@
+package lt.vilnius.tvarkau.mvp.interactors
+
+import lt.vilnius.tvarkau.backend.LegacyApiService
+import lt.vilnius.tvarkau.backend.requests.GetReportTypesRequest
+import rx.Scheduler
+import rx.Single
+
+/**
+ * @author Martynas Jurkus
+ */
+class ReportTypesInteractorImpl(
+        val api: LegacyApiService,
+        val ioScheduler: Scheduler
+) : ReportTypesInteractor {
+
+    private val entryCache = mutableListOf<String>()
```

Comment:
If list not modified: items added, removed. I would prefer to have immutable version with mutable variable